src/libs/TransactionUtils.js

```diff
@@ -309,7 +309,8 @@ function waypointHasValidAddress(waypoint) {
  * @returns {Object} validated waypoints
  */
 function getValidWaypoints(waypoints, reArrangeIndexes = false) {
-    const waypointValues = _.values(waypoints);
+    const sortedIndexes = _.map(_.keys(waypoints), (key) => parseInt(key.replace('waypoint', ''), 10)).sort();
```

Comment:
There are multiple places where this logic is used. Can we make util function?
There are 3 usages of `const index = Number(key.replace('waypoint', ''));`

Reply:
agree, and we eventually have to change the index logic for waypoints...